### ISSUE TITLE
Fatal error if InAppBrowser is close immediately

### ISSUE THREAD

**Issue by lakshmana** [apache/cordova-plugin-inappbrowser#368]
I have this error when the inAppBrowser is close soon enough ( for eg within  1 sec )

> `java.lang.NullPointerException: Attempt to invoke virtual method 'void org.apache.cordova.inappbrowser.InAppBrowserDialog.show()' on a null object reference`

I could solve the problem by adding a null check on line no. 282 on inAppBrowser.java as follows
replace `dialog.show();` with `if(dialog != null) dialog.show();`

I am not an expert on java android and so can someone confirm this fix or give a good way to fix the problem.

Thx

**Comment by dpa99c** [comment]
There is already some defensive coding around async invocation of dialog methods such as [when dismissing on close](https://github.com/apache/cordova-plugin-inappbrowser/blob/master/src/android/InAppBrowser.java#L523) but these not-null checks should also be added to where the dialog is being explicitly [shown](https://github.com/apache/cordova-plugin-inappbrowser/blob/master/src/android/InAppBrowser.java#L303) and [hidden](https://github.com/apache/cordova-plugin-inappbrowser/blob/master/src/android/InAppBrowser.java#L314) because these operations are running on the UI thread, so there is the possibility of such concurrency issues due to the race condition of the dialog instance being destroyed on the another thread before it is invoked on the UI thread.

**Comment by dpa99c** [comment]
@lakshmana By which I mean doing what you said: `if(dialog != null){ dialog.show(); }`

**Comment by lakshmana** [comment]
yes, it seems to be a case of race condition and I also confirm that no called were made to the dialog from my code. The dialog could have been destroyed by the time the code dialog.show() was invoked.

**Comment by boynet** [comment]
I am also getting lot of errors in my analytics around this line:
```

java.lang.NullPointerException
org.apache.cordova.inappbrowser.InAppBrowser$2.run

  at org.apache.cordova.inappbrowser.InAppBrowser$2.run (InAppBrowser.java:282)
  at android.os.Handler.handleCallback (Handler.java:873)
  at android.os.Handler.dispatchMessage (Handler.java:99)
  at android.os.Looper.loop (Looper.java:201)
  at android.app.ActivityThread.main (ActivityThread.java:6806)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:547)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:873)
```

or

```
  at org.apache.cordova.inappbrowser.InAppBrowser$2.run (InAppBrowser.java:282)
  at android.os.Handler.handleCallback (Handler.java:789)
  at android.os.Handler.dispatchMessage (Handler.java:98)
  at android.os.Looper.loop (Looper.java:164)
  at android.app.ActivityThread.main (ActivityThread.java:6940)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.Zygote$MethodAndArgsCaller.run (Zygote.java:327)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1374)
```

**Comment by NiklasMerz** [comment]
This issue happens for me if `show()` is called after creating an IAB for target "_system". 

Is this issue still worth a PR? Should I add null checks in async show and hide to avoid issues like this?
